python/cudf/cudf/core/groupby/groupby.py

```diff
@@ -275,6 +275,44 @@ def __iter__(self):
         for i, name in enumerate(group_names):
             yield name, grouped_values[offsets[i] : offsets[i + 1]]
 
+    @property
+    def dtypes(self):
+        """
+        Return the dtypes in this group.
+
+        Returns
+        -------
+        pandas.DataFrame
+            The data type of each column of the group.
+
+        Examples
+        --------
+        >>> import cudf
+        >>> df = cudf.DataFrame({'a': [1, 2, 3, 3], 'b': ['x', 'y', 'z', 'a'],
+        ...                      'c':[10, 11, 12, 12]})
+        >>> df.groupby("a").dtypes
+                b      c
+        a
+        1  object  int64
+        2  object  int64
+        3  object  int64
+        """
+        non_grouped = [
+            name
+            for name in self.obj._data.names
+            if name not in self.grouping.names
+        ]
```

Comment:
I think this does the same:

```suggestion
        non_grouped = self.grouping.values._column_names
```